js/runner.go

```diff
@@ -312,6 +307,77 @@ func (r *Runner) IsExecutable(name string) bool {
 	return exists
 }
 
+// HandleSummary calls the specified summary callback, if supplied.
+func (r *Runner) HandleSummary(ctx context.Context, summary *lib.Summary) (map[string]io.Reader, error) {
+	summaryDataForJS := summarizeMetricsToObject(summary, r.Bundle.Options)
+
+	out := make(chan stats.SampleContainer, 100)
+	defer close(out)
+
+	go func() { // discard all metrics
+		for range out {
+		}
+	}()
+
+	vu, err := r.newVU(0, out)
+	if err != nil {
+		return nil, err
+	}
+
+	handleSummaryFn := goja.Undefined()
+	if exported := vu.Runtime.Get("exports").ToObject(vu.Runtime); exported != nil {
+		fn := exported.Get(consts.HandleSummaryFn)
+		if _, ok := goja.AssertFunction(fn); ok {
```

Comment:
Sorry, just realized I've forgotten to respond here. I am not sure if what you suggest will work, since we're creating a new VU here and `vu.Runtime.Get("exports")` would be different than `r.Bundle.exports`. Moreover, this way we'll raise an error if `handleSummary` was exported but it wasn't a function.